core/src/main/java/org/apache/iceberg/rest/requests/FetchScanTasksRequest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.requests;
+
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.rest.RESTRequest;
+import org.immutables.value.Value;
+
+@Value.Immutable
```

Comment:
I would prefer to stop using Immutables. This library has caused quite a few annoying issues and affected how code is structured. Especially for simple objects like this one it doesn't seem worth the headache of using generated code.

Reply:
I can switch back to not using Immutables. The reason I opted was because I saw some of the iceberg rest request classes using it.

Examples:
https://github.com/apache/iceberg/blob/09370ddbc39fc3920fb8cbd3dff11b377dd37e40/core/src/main/java/org/apache/iceberg/rest/requests/CreateViewRequest.java

https://github.com/apache/iceberg/blob/09370ddbc39fc3920fb8cbd3dff11b377dd37e40/core/src/main/java/org/apache/iceberg/rest/requests/ReportMetricsRequest.java